.github/workflows/unit_test.yml

```
@@ -15,15 +15,15 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
 
-      - name: install dependencies
+      - name: Install Dependencies
         run: pip install -r REQUIREMENTS.txt
 
-      - name: tests
-        run: pytest
-
-      - name: black formatting check
+      - name: Black Formatting Check
         run: black --check .
 
+      - name: Run Tests & Generate Report
+        run: coverage run -m unittest tests/test_game.py
+
```

The order isn't the most important thing, but I want to have the unit tests run before the style.  

I think I've figured out the coverage issue, and I'd like to continue using pytest.  I'll put some other comments in that should describe how to fix it.

```suggestion
      - name: Run Tests & Generate Report
         run: pytest --cov
         
      - name: Black Formatting Check
        run: black --check .
```